internal/worker/json.go

```diff
@@ -640,6 +640,12 @@ type ManifestInfo struct {
 	// List of relevant modules used by osbuild-composer which
 	// could affect the manifest content.
 	OSBuildComposerDeps []*OSBuildComposerDepModule `json:"osbuild_composer_deps,omitempty"`
+
+	// Ordered list of pipeline names in the manifest, separated into build and
+	// payload pipelines. These are parsed from the manifest itself and copied
+	// to the osbuild job result so it can properly order the osbuild job log
+	// by pipeline execution order.
+	PipelineNames *PipelineNames `json:"pipeline_names,omitempty"`
```

Comment:
It should be possible to simplify the flow and use this field even for manifests generated on the server side by composer. Something to consider for the future...

Reply:
What I meant is that we could be setting this field in the `ManifestInfo` even when generating the manifest on the server side, instead of adding he information directly to the `osbuild` job arguments. This way, we would have a single place for it (in the `ManifestInfo`) and unify it, regardless of how we generated the manifest.

I didn't mean that we should parse it from the manifest when we are generating the manifest on the server side. We should, however add it to the `ManifestInfo`, instead of the `osbuild` job args.

Does this make more sense now?